.github/workflows/native.yml

```diff
@@ -0,0 +1,49 @@
+name: native
+
+on:
+  push:
+    branches:
+      - main
+      - master
+  pull_request:
+    branches:
+      - main
+      - master
+
+jobs:
+  changes:
+    runs-on: ubuntu-latest
+    outputs:
+      backend: ${{ steps.filter.outputs.backend }}
+      native: ${{ steps.filter.outputs.native }}
+    steps:
+      - uses: actions/checkout@v4
+      - uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          filters: |
+            backend:
+              - 'backend/**'
+              - '.github/workflows/**'
+            native:
+              - 'native/**'
+              - '.github/workflows/**'
+
+  test-app:
+    name: Test app
+    needs: changes
+    if: ${{ needs.changes.outputs.native == 'true' }}
+    runs-on: ubuntu-latest
+    timeout-minutes: 7
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v3
+        with:
+          node-version: 14
```

Comment:
Let's go ahead and bump this to the latest version of node supported by react native or our docker-compose. 18 it seems.

We also don't need to set the `npm` version 7 since this is a new project we can use whatever latest version is shipped with node 18

Reply:
Done